gradio/components/dataframe.py

```diff
@@ -401,7 +401,7 @@ def get_metadata(
 
         if isinstance(value, Styler):
             return Dataframe.__extract_metadata(
```

Comment:
Was there also a bug with `hidden_columns` how do we test this?

Reply:
No, actually it was fine originally but the first time I refactored this, I introduced a bug that was caught by a preexisting unit test: `pytest test/components/test_dataframe.py::TestDataframe::test_dataframe_hidden_columns`

Reply:
Its a subtle bug so I added the inline comments